pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -964,6 +977,42 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 
 		domain.Spec.Devices.Disks = append(domain.Spec.Devices.Disks, newDisk)
 	}
+	// Handle virtioFS
+	for _, fs := range vmi.Spec.Domain.Devices.Filesystems {
+		if fs.Virtiofs != nil {
+			newFS := FilesystemDevice{}
+
+			newFS.Type = "mount"
+			newFS.AccessMode = "passthrough"
+			newFS.Driver = &FilesystemDriver{
+				Type:  "virtiofs",
+				Queue: "1024",
+			}
+			newFS.Binary = &FilesystemBinary{
+				Path:  "/usr/libexec/virtiofsd",
+				Xattr: "on",
+				Cache: &FilesystemBinaryCache{
+					Mode: "always",
+				},
+				Lock: &FilesystemBinaryLock{
+					Posix: "on",
+					Flock: "on",
+				},
+			}
+			newFS.Target = &FilesystemTarget{
+				Dir: fs.Name,
+			}
+
+			volume := volumes[fs.Name]
+			if volume == nil {
+				return fmt.Errorf("No matching volume with name %s found", fs.Name)
```

Comment:
i think it's fine to keep this check here. It would be good to add this to the validation webhook as well though so we can catch this condition earlier.

i don't think the PR necesssarily needs to be blocked for that though. it can come in a follow up pr